final_task/rss_reader/book_gen.py

```diff
@@ -0,0 +1,110 @@
+import time
+import os.path
+import html
+
+from ebooklib import epub
+
+DATE_FORMAT = "%Y%m%d"
+
+
+def _render_document(title, items, date_str):
+    html = ["<html><head><title>"]
+    h_title = f"{date_str}"
+    html.append(h_title)
+    html.append("</title></head><body>")
+
+    html.append("<h1>")
+    html.append(title)
+    html.append("</h1>")
+
+    for i in items:
+        html.append(_render_item(i))
+
+    html = "".join(html)
+    return html
+
+
+def _render_item(item):
+    html = ["<hr>"]
+    title = item["title"] or "No Headline"
+    html.append(f"<h2>{title}</h2>")
+    date = time.strftime("%a, %d %b %Y %H:%M:%S", item["date"])
+    html.append(f"<p><i>{date}</i></p>")
+    if item["link"] is not None:
+        link = item["link"]
+        html.append(f"<p><a href='{link}'>{link}</a></p>")
+    else:
+        html.append(f"<p><a>no link</a></p>")
+    description = item["description"] or "No description"
+    html.append(f"<p>{description}</p>")
+    html = "".join(html)
+    return html
+
+
+def _gen_id(title: str, date:str) -> str:
+    """
+    generate string for book id and file name
+
+    :return: generated string
+    """
+    string = title + "_" + date
+    string = string.lower()
+    for c in r'\|/:*?"<>':
+        string.replace(c, '_')
+    if len(string) > 122:
+        string = string[:121] + "..."
+    return string
+
+
+def _create_html(book_id, bookpath, text):
+    # text = html.escape(text)
+    file = None
+    if os.path.isdir(bookpath):
+        file = open(os.path.join(bookpath, book_id + ".html"), "w", encoding="utf-8")
+    elif not os.path.exists(bookpath) or os.path.isfile(bookpath) or os.path.islink(bookpath):
+        file = open(bookpath, "w", encoding="utf-8")
+    try:
+        file.write(text)
+        file.close()
+    except (AttributeError, OSError):
+        pass
```

Comment:
Usually if you leave `pass` in the `except` block it is better to leave a comment why.